ZPastel.Persistence/Configuration/OrderConfiguration.cs

```diff
@@ -13,15 +13,14 @@ public void Configure(EntityTypeBuilder<Order> builder)
             builder.HasKey(o => o.Id);
 
             builder.Property(o => o.Id).HasColumnName("Id");
-            builder.Property(o => o.TotalPrice).HasColumnName("TotalPrice");
-            builder.Property(o => o.CreatedOn).HasColumnName("CreatedOn");
+            builder.Property(o => o.TotalPrice).HasColumnName("TotalPrice").HasColumnType("decimal(18,2)");
             builder.Property(o => o.CreatedByUsername).HasColumnName("CreatedByUsername");
             builder.Property(o => o.CreatedById).HasColumnName("CreatedById");
             builder.Property(o => o.CreatedOn).HasColumnName("CreatedOn");
             builder.Property(o => o.LastModifiedById).HasColumnName("LastModifiedById");
             builder.Property(o => o.LastModifiedOn).HasColumnName("LastModifiedOn");
 
-            builder.HasMany(o => o.OrderItems).WithOne().HasForeignKey(e => e.OrderId);
+            builder.HasMany(o => o.OrderItems).WithOne().HasForeignKey(e => e.OrderItemId).Metadata.DeleteBehavior = DeleteBehavior.Restrict;
```

Comment:
Create a delete scenario to cover this config

Reply:
When deleting an Order we should delete its items as well

Reply:
Validate the changes querying directly the in memory DBContext in test layer

Reply:
Created delete scenario
Validated changes